src/game/systems/input.rs

```diff
@@ -1,15 +1,22 @@
 use super::*;
-use crate::game::{Camera, Player, Position};
+use crate::game::{Camera, Player, Position, Target, TargetPosition, Velocity};
 
 pub struct InputSystem;
 
 impl InputSystem {
     pub fn control_player(
-        mut player: UniqueViewMut<Player>,
+        player: UniqueView<Player>,
         positions: View<Position>,
-        camera: UniqueView<Camera>
+        camera: UniqueView<Camera>,
+        mut velocities: ViewMut<Velocity>,
+        mut targets: ViewMut<Target>,
+        mut target_positions: ViewMut<TargetPosition>,
     ) {
-        player.vel.0 = if is_key_down(KeyCode::W) || is_key_down(KeyCode::Up) {
+        let player_vel = &mut velocities[player.0];
```

Comment:
Indexing a view is the same as `get(id).unwrap()`.
It didn't look like the game could work without the player present. But if it can then all indexing should change to `get`.

Reply:
Yea the game is intended to only work with a player present. So I'll keep it indexing views then!